sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemResponse.java

```diff
@@ -28,26 +33,60 @@
 public class CosmosItemResponse<T> {
     private final Class<T> itemClassType;
     private final ItemDeserializer itemDeserializer;
-    byte[] responseBodyAsByteArray;
+
     //  Converting item to volatile to fix Double-checked locking - https://en.wikipedia.org/wiki/Double-checked_locking
     //  http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html
     private volatile T item;
+    private volatile JsonNode itemBodyOverride;
     final ResourceResponse<Document> resourceResponse;
     private InternalObjectNode props;
 
-    private AtomicBoolean hasTrackingIdCalculated = new AtomicBoolean(false);
+    private final AtomicBoolean hasTrackingIdCalculated = new AtomicBoolean(false);
 
     private boolean hasTrackingId;
 
+    private final Supplier<Boolean> hasPayload;
+
     CosmosItemResponse(ResourceResponse<Document> response, Class<T> classType, ItemDeserializer itemDeserializer) {
-        this(response, response.getBodyAsByteArray(), classType, itemDeserializer);
+        this.itemClassType = classType;
+        this.resourceResponse = response;
+        this.itemDeserializer = itemDeserializer;
+        this.item = null;
+        this.hasPayload = () -> response.hasPayload();
+        this.itemBodyOverride = null;
     }
 
-    CosmosItemResponse(ResourceResponse<Document> response, byte[] responseBodyAsByteArray, Class<T> classType, ItemDeserializer itemDeserializer) {
+    private CosmosItemResponse(ResourceResponse<Document> response, T item, JsonNode itemBodyOverride, Class<T> classType, ItemDeserializer itemDeserializer) {
         this.itemClassType = classType;
-        this.responseBodyAsByteArray = responseBodyAsByteArray;
         this.resourceResponse = response;
         this.itemDeserializer = itemDeserializer;
+        this.item = item;
+        this.itemBodyOverride = itemBodyOverride;
+        boolean hasPayloadStaticValue = item != null;
+        this.hasPayload = () -> hasPayloadStaticValue;
+        this.itemBodyOverride = null;
```

Comment:
line 64 and line 67 both are trying to set the `itemBodyOverride`, what is the use case for `itemBodyOverride`

Reply:
Used from Encryption - when the modified JsonNode needs to be kept after decryption/encryption

Reply:
Fixed